tests/unit/s2n_mem_usage_test.c

```diff
@@ -162,6 +181,8 @@ int main(int argc, char **argv)
     fprintf(stdout, "VmData initial:           %10zu\n", vm_data_initial);
     fprintf(stdout, "VmData after allocations: %10zu\n", vm_data_after_allocation);
     fprintf(stdout, "VmData after handshakes:  %10zu\n", vm_data_after_handshakes);
+    fprintf(stdout, "VmData after free:        %10zu\n", vm_data_after_free);
+    fprintf(stdout, "VmData after release:     %10zu\n", vm_data_after_release_buffers);
     fprintf(stdout, "Max VmData diff allowed:  %10zu\n", MAX_ALLOWED_MEM_DIFF);
 #endif
 
```

Comment:
Do we need any expectations on the size after buffers were released?

Reply:
unfortunately, this doesn't really change much in this test.  i think we just set expectations that they are not -1.

Reply:
So, in rev-2 i did add some expectations, however the test would fail in some scenarios.  For example, in FIPS mode, we don't see any reduction in memory after release buffers.

So as a compromise I've simply made a check that we don't grow our vm size after free and release.